src/pathpyG/core/Graph.py

```diff
@@ -403,19 +399,9 @@ def __setitem__(self, key: str, val: torch.Tensor) -> None:
             else:
```

Comment:
Does this method really do what it is supposed to do? If I understand it correctly, you can only change graph level attributes but not assign new ones due to `if key in self.data.keys()`. For node and edge level attributes it throws `AttributeError: 'Graph' object has no attribute 'key'`. It is probably meant to be `key[0].starts_...` without `self.`. But then it would be possible to assign new attributes for a single node which does not work in `pytorch_geometric`.

If this method should only be able to change existing attributes then we should add a similar check when `if instance(key. tuple)`. If it should also be able to assign new attributes at node and edge level, then we need to think of a default value to assign to the other nodes when the attribute is only assigned to one node or edge. Additionally, there should maybe be a way to add attributes for all nodes and edges with a single tensor at once.